src/Symfony/Component/Lock/LockFactory.php

```diff
@@ -19,4 +19,14 @@
  */
 class LockFactory extends Factory
 {
+    /**
+     * Creates a lock for the given resource.
+     *
+     * @param float|null $ttl         Maximum expected lock duration in seconds
+     * @param bool       $autoRelease Whether to automatically release the lock or not when the lock instance is destroyed
+     */
+    public function createLock($resource, $ttl = 300.0, $autoRelease = true): Lock
```

Comment:
Could you add type declarations?

And the $resource param is not explained in the php doc

Reply:
`$resource` phpdoc re-added. For type declarations, they are added in #32555 (the class extends the deprecated `Factory` on 4.4, adding them there would cause a signature mismatch)

Reply:
Thanks 👍🏻